server/datastore/redis/redistest/redistest.go

```diff
@@ -35,7 +35,7 @@ func NopRedis() fleet.RedisPool {
 	return nopRedis{}
 }
 
-func SetupRedis(tb testing.TB, cluster, redir, readReplica bool) fleet.RedisPool {
+func SetupRedis(tb testing.TB, cleanupKeyPrefix string, cluster, redir, readReplica bool) fleet.RedisPool {
```

Comment:
This change is required because we used to remove all keys from redis in the test cleanup, but as we add more redis-based tests in multiple packages, this means the tests run in parallel, access the same redis server, and see their keys disappearing from under them right in the middle of tests, causing random failures.

This approach forces us to more carefully write our redis tests, so that all keys in the test use a common prefix, making it isolated from other tests and possible to selectively remove those keys on cleanup. Another approach would be to launch distinct redis for each tests (not really practical) or use in-memory mocks of redis (would not allow us to gain confidence e.g. in standalone vs cluster mode).

I think this selective cleanup approach is good for now (I managed to find proper prefixes for all cases, not deleting anything in some of them as it didn't matter), we can see in the long term if that's manageable.